### PR TITLE
Fixed corelist -r param processing bug for 0 and non-number values

### PR DESCRIPTION
**Pull request by kozachkov**
These cases was failed:

```
corelist -r 0 # works like corelist -r that's not right
corelist -r a # show the warning: Argument "a" isn't numeric in addition (+) at
```

### CODE REVIEW

File: dist/Module-CoreList/corelist [hunk 1]
```diff
@@ -498,11 +499,11 @@ sub format_perl_version {
 sub numify_version {
     my $ver = shift;
     if ($ver =~ /\..+\./) {
-	have_version_pm()
-	    or die "You need to install version.pm to use dotted version numbers\n";
+        have_version_pm()
```
Comment (Leont) [inline dist/Module-CoreList/corelist:502]: This commit seems to contain a lot of unrelated whitespace changes

Comment (kozachkov) [reply dist/Module-CoreList/corelist:502]: @Leont not so many actually just what I saw when was fixing the bug. Nevertheless that's not good?

Comment (Leont) [reply dist/Module-CoreList/corelist:502]: Doing that in the same commit where you're making semantic changes is always bad IMNSHO.

Doing it in a separate commit may or may not be desired, I think that's up to the maintainer of Module::CoreList (@bingos), but I would generally recommend against doing that unasked.

Comment (kozachkov) [reply dist/Module-CoreList/corelist:502]: @Leont Thank you. I will know for the future changes.

**Comment by jkeenan** [comment]
@khml, I believe your proposed fix for `corelist -r 0` is satisfactory, but your discussion of `corelist -r a` reveals more subtle problems with `corelist`.

If you would like to pursue this pull request, please force push a patch without the whitespace changes.   Otherwise, we will probably close this p.r.

Thank you very much.
Jim Keenan

**Comment by jkeenan** [comment]
> @khml, I believe your proposed fix for `corelist -r 0` is satisfactory, but your discussion of `corelist -r a` reveals more subtle problems with `corelist`.
> 
> If you would like to pursue this pull request, please force push a patch without the whitespace changes. Otherwise, we will probably close this p.r.
> 

There's been no response in nearly 12 months.  I am taking this p.r. for the purpose of closing it within 7 days unless there is a good argument for keeping it open.

**Comment by jkeenan** [comment]
> > @khml, I believe your proposed fix for `corelist -r 0` is satisfactory, but your discussion of `corelist -r a` reveals more subtle problems with `corelist`.
> > If you would like to pursue this pull request, please force push a patch without the whitespace changes. Otherwise, we will probably close this p.r.
> 
> There's been no response in nearly 12 months. I am taking this p.r. for the purpose of closing it within 7 days unless there is a good argument for keeping it open.

Closing as per schedule.
